presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -481,9 +487,9 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
                             IcebergColumnHandle::getName,
                             IcebergColumnHandle::getType)));
         }).orElseGet(() -> {
-            if (!mergeStrategy.equals(HiveStatisticsMergeStrategy.NONE)) {
-                PartitionStatistics hiveStats = metastore.getTableStatistics(getMetastoreContext(session), handle.getSchemaName(), handle.getIcebergTableName().getTableName());
-                return mergeHiveStatistics(icebergStatistics, hiveStats, mergeStrategy, icebergTable.spec());
+            PartitionStatistics hiveStats = metastore.getTableStatistics(getMetastoreContext(session), handle.getSchemaName(), handle.getIcebergTableName().getTableName());
+            if (!mergeFlags.isEmpty()) {
+                return mergeHiveStatistics(icebergStatistics, hiveStats, mergeFlags, icebergTable.spec());
```

Comment:
Should we get hive stats inside the `if` clause?

Reply:
Yup. Thanks for catching it